elastic/kibana
# Better handling of Monitoring in Elastic Cloud

## Summary

This resolves https://github.com/elastic/kibana/issues/43260

A new message is now displayed instead of the old one when we are running in cloud and there is a remote exporter configured but no local exporter.

The old message looked like this:
![62648214-3369eb80-b907-11e9-8a3d-6c229f82b5c3](https://user-images.githubusercontent.com/111616/63169110-80477380-c025-11e9-9941-565ffb273ec2.png)


The new message looks like this:

<img width="602" alt="Screen Shot 2019-08-16 at 14 50 42" src="https://user-images.githubusercontent.com/111616/63168995-36f72400-c025-11e9-88fa-444efeb33ab0.png">

## Testing
Testing this PR is slightly awkward. I don't know if there is a way to test this *in* a cloud environment itself, but maybe @pickypg or @tylersmalley can provide guidance? 

Outside of that, the best way I can think of to test this is as follows:

1) Bring up Elasticsearch and Kibana instances. Do not enable Monitoring.

2) Issue the following:

```
PUT /_cluster/settings
{
    "transient" : {
        "xpack.monitoring.exporters" : {
            "my_remote": {
                "type": "http",
                "host": [ "10.1.2.3:9200"]
            }
        }
    }
}
```

3) Observe the old error message shown at the top of this PR.

4) Modify L94 in find_reason.js to `if (isCloud) {`.

5) Reload the Monitoring application and observe the new error message.



### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
